### ISSUE TITLE
DBus prefix of data server breaks integration with calender etc.

### ISSUE THREAD

**Issue by fmorgner** [flathub/org.gnome.Evolution#91]
After migrating to a more flatpak oriented installation, I was surprised to notice that Gnome Calendar (also flatpak) and Gnome Contacts (flatpak as well) don't see the information provided by the evolution data server anymore. I suspect it might be related to the fact the DBus names are prefixed with "org.gnome.Evolution" meaning the services are now called "org.gnome.Evolution.org.gnome.evolution.dataserver.*". What is the reason for this prefix being introduced in the first place, and would it be possible to get rid of it?

**Comment by fcrozat** [comment]
Unfortunately, this change also break integration of e-d-s calendar events from flatpak to be displayed by gnome-shell calendar which is running outside of flatpak.

**Comment by Sesivany** [comment]
It's not a change. Evolution flatpak has worked this way from the very beginning. A flatpak app should not rely so greatly on something on the host. Unless someone creates portals with stable API that will provide calendar, contacts, tasks data which we can safely rely on it's not going to change.

**Comment by fmorgner** [comment]
I'm not sure I follow entirely. Since e-d-s is bundled with the Evolution flatpak, why would there be system (non-flatpak) version of it running on a system where one chooses to go the flatpak route? I can accept that this was a design decision, but I'd like to fully understand it.

@Sesivany regarding Evolution flatpak working like this from the beginning, I am also confused. Wasn't the e-d-s dbus prefix introduced with https://github.com/flathub/org.gnome.Evolution/commit/0e18ea6a76a8405007839e8d5678e8c570b9955b?

**Comment by mcrha** [comment]
There is not considered (also intentionally) to expose the eds D-Bus interface from the Flatpak app to the host system, which could override the host system D-Bus services, thus either a Flatpak app connects to the host system eds (and deals with its bugs), or it bundles the eds into itself and brings clear experience with known fixed bugs to the user.

> regarding Evolution flatpak working like this from the beginning, I am also confused. Wasn't the e-d-s dbus prefix introduced with [0e18ea6](https://github.com/flathub/org.gnome.Evolution/commit/0e18ea6a76a8405007839e8d5678e8c570b9955b)?

There had been multiple attempts to make it right. It was an iterative process. Nonetheless the idea to bundle also the eds D-Bus processes into the Evolution Flatpak was clear since the beginning. Reasons given in https://github.com/flathub/org.gnome.Evolution/issues/91#issuecomment-1111050704 .

**Comment by fcrozat** [comment]
> It's not a change. Evolution flatpak has worked this way from the very beginning. A flatpak app should not rely so greatly on something on the host. Unless someone creates portals with stable API that will provide calendar, contacts, tasks data which we can safely rely on it's not going to change.

There is a misunderstanding here. I'm not saying the flatpak app should rely on the host but the contrary.
Gnome-shell-calendar is shipped on the host and pulls by dependendy e-d-s on the host. And since e-d-s on the host and e-d-s on flatpak have different dbus prefix (and flatpak e-d-s is not replacing the host e-d-s), gnome-shell can't display calendar entries.

Maybe a solution would be to move gnome-shell-calendar to evolution and ship it as part of evolution flatpak (or in a add-on)

**Comment by mcrha** [comment]
> gnome-shell can't display calendar entries.

That's true. It's the same for any other flatpak or the host system application, which uses the evolution-data-server. Just as you said, because the Evolution flatpak application doesn't expose these D-Bus interfaces in the standard names. It's intentional, I'm sorry.

A real "fix" for this would be to have an evolution-data-server Flatpak "runtime", which could replace the system D-Bus services and to which the Flatpak apps using it could connect. It's not going to happen, at least not any time soon, even I agree it could be an interesting functionality for the related parties.
